suplemon/viewer.py

```diff
@@ -50,76 +51,38 @@ def __init__(self, app, window):
         self.x_scroll = 0
         self.cursors = [Cursor()]
 
-        # Lexer for translating tokens to strings
-        self.lexer = None
-        # Built in syntax definition (for commenting etc.)
-        self.syntax = None
-        # Normal Pygments lexer
-        self.pygments_syntax = None
+        # Copy/paste buffer
+        self.buffer = []
+
+        # Runnable methods
+        self.operations = {
+            "arrow_right": self.arrow_right,              # Arrow Right
+            "arrow_left": self.arrow_left,                # Arrow Left
+            "arrow_up": self.arrow_up,                    # Arrow Up
+            "arrow_down": self.arrow_down,                # Arrow Down
+            "jump_left": self.jump_left,                  # Ctrl + Left
+            "jump_right": self.jump_right,                # Ctrl + Right
+            "jump_up": self.jump_up,                      # Ctrl + Up
+            "jump_down": self.jump_down,                  # Ctrl + Down
+            "page_up": self.page_up,                      # Page Up
+            "page_down": self.page_down,                  # Page Down
+            "home": self.home,                            # Home
+            "end": self.end,                              # End
+            "find": self.find_query                       # Ctrl + F
+        }
```

Comment:
The purpose of the abstraction would be to reduce the maintenance and make the functions/commands consistent (e.g. `find` -> `find`, no renaming). Otherwise, every time you add a new operation, you will be adding a new key. This isn't a lot to start but it's overwhelming when there's 50 keys.

Upon further consideration, I think an `Operation` class is probably easier to manage since it can be placed in a separate file.

That being said, the decorator implementation would look like (I realized that saving to a dict is much saner than marking/searching functions):

```python
operations = {}

def operation(fn):
    """Save our function as an operation"""
    operations[fn.__name__] = fn

class Viewer(object):
    def run_operation(self, operation):
      # Simplified for example
      return operations[operation](self)

    @operation
    def arrow_right(self):
      pass
```

For renaming, we would define a proxy operation:

```python
operations = {}

def operation(name):
    """Save our function as an operation"""
    def operation_proxy(fn):
      operations[name] = fn
    return operation_proxy

class Viewer(object):
    def run_operation(self, operation):
      # Simplified for example
      return operations[operation](self)

    @operation("arrow_right")
    def arrow_right(self):
      pass
```